pkg/network/tracer/tracer.go

```diff
@@ -161,48 +153,42 @@ func newTracer(cfg *config.Config) (*Tracer, error) {
 		}
 	}
 
-	ebpfTracer, err := connection.NewTracer(cfg, bpfTelemetry)
+	tr := &Tracer{
+		config:                     cfg,
+		lastCheck:                  atomic.NewInt64(time.Now().Unix()),
+		sysctlUDPConnTimeout:       sysctl.NewInt(cfg.ProcRoot, "net/netfilter/nf_conntrack_udp_timeout", time.Minute),
+		sysctlUDPConnStreamTimeout: sysctl.NewInt(cfg.ProcRoot, "net/netfilter/nf_conntrack_udp_timeout_stream", time.Minute),
+	}
+	defer func() {
+		if reterr != nil {
+			tr.Stop()
+		}
+	}()
+
+	if nettelemetry.EBPFTelemetrySupported() {
+		tr.bpfTelemetry = nettelemetry.NewEBPFTelemetry()
+		coretelemetry.GetCompatComponent().RegisterCollector(tr.bpfTelemetry)
+	}
+
+	tr.ebpfTracer, err = connection.NewTracer(cfg, tr.bpfTelemetry)
 	if err != nil {
 		return nil, err
 	}
-	coretelemetry.GetCompatComponent().RegisterCollector(ebpfTracer)
+	coretelemetry.GetCompatComponent().RegisterCollector(tr.ebpfTracer)
 
-	conntracker, err := newConntracker(cfg, bpfTelemetry)
+	tr.conntracker, err = newConntracker(cfg, tr.bpfTelemetry)
```

Comment:
Can `tr.conntracker` be `nil` here?

Reply:
What do you mean? It is being assigned to.

Reply:
You check for `tr.conntracker != nil` in `Stop()`, but assume that is going to be `non-nil` after this line. Can `tr.conntracker` ever be non-`nil` if we don't return an error from `newTracer`?

Reply:
if `newConntracker` returns an error, `tr.conntracker` will be `nil`.